packages/block-library/src/navigation/editor.scss

```diff
@@ -73,6 +73,20 @@ $navigation-item-height: 46px;
 	}
 }
 
+// Move Separators to before element cause after is used for outlining the block.
+.wp-block-navigation > .wp-block-navigation__container > .wp-block-navigation-link {
```

Comment:
If we move ::before to `wp-block-navigation-link__label`, then we might be able to have better control over its placement within the block's foot print. I tried to hack it in, and kind of get it working, but then it breaks the `data-separator`.